### PR TITLE
Support actions with MySQL and MariaDB

### LINKED ISSUE #26478: [P2] Bring actions to MySQL and MariaDB

### PR DESCRIPTION
**Pull request by metamben**
Fixes #26478.

### CODE REVIEW

File: test/metabase/api/dashboard_test.clj [hunk 1]
```diff
@@ -2366,26 +2366,26 @@
   {:is_write true
    :parameters [{:id field-name :slug field-name :target ["variable" ["template-tag" field-name]] :type :text}]
    :dataset_query (mt/native-query
-                    {:query (format "insert into TYPES (%s) values ({{%s}})"
+                    {:query (format "insert into types (%s) values ({{%s}})"
```
Comment (metamben) [inline test/metabase/api/dashboard_test.clj:2369]: MySQL is case sensitive.

File: test/metabase/api/dashboard_test.clj [hunk 2]
```diff
@@ -2366,26 +2366,26 @@
...
-                 {:field-name "xml" :base-type :type/XML ::good "<xml></xml>"}
-                 {:field-name "boolean" :base-type :type/Boolean ::good true ::bad "not boolean"}
-                 {:field-name "integer" :base-type :type/Integer ::good 100}
-                 {:field-name "float" :base-type :type/Float ::good 0.4}
+    (let [types [{:field-name "atext" :base-type :type/Text ::good "hello"}
+                 {:field-name "ajson" :base-type :type/JSON ::good "{\"a\": 1}"}
+                 {:field-name "axml" :base-type :type/XML ::good "<xml></xml>"}
+                 {:field-name "aboolean" :base-type :type/Boolean ::good true ::bad "not boolean"}
+                 {:field-name "ainteger" :base-type :type/Integer ::good 100}
+                 {:field-name "afloat" :base-type :type/Float ::good 0.4}
```
Comment (metamben) [inline test/metabase/api/dashboard_test.clj:2381]: In this test we don't intend to check how the drivers can deal with reserved keywords.

File: test/metabase/actions/test_util.clj [hunk 3]
```diff
@@ -126,7 +126,7 @@
                                            :dataset_query {:database (mt/id)
                                                            :type     :native
                                                            :native   {:query         (str "UPDATE categories\n"
-                                                                                          "SET name = [[{{name}} || ' ' ||]] 'Shop'\n"
+                                                                                          "SET name = concat([[{{name}}, ' ', ]] 'Sh', 'op')\n"
```
Comment (metamben) [inline test/metabase/actions/test_util.clj:129]: MySQL doesn't support the `||` operator.

**Review by snoe — APPROVED** [review]
LGTM

**Pull request merged**